From 38ad6c8d75fa9e0f3ebcc6320272dd5252505c5f Mon Sep 17 00:00:00 2001
From: Marco Capetta <mcapetta@sipwise.com>
Date: Thu, 29 Dec 2022 10:14:25 +0100
Subject: MT#55831 Handle BYE after 180 with to_tag

- customer has a scenario when 180 has to_tag and call terminated by caller with BYE but calle did not get any call termination signalling.
- a bit more special for such cases BYE handling added.
(real ticket number: TT#73957)

--- a/apps/sbc/CallLeg.cpp
+++ b/apps/sbc/CallLeg.cpp
@@ -961,7 +961,11 @@ void CallLeg::onSipRequest(const AmSipRe
     }
   }
   else {
-    if(getCallStatus() == Disconnected &&
+    if (getCallStatus() == Ringing && !getOtherId().empty() && req.method == SIP_METH_BYE) {
+        dlg->reply(req,200,"OK");
+        stopCall(&req);
+    }
+    else if(getCallStatus() == Disconnected &&
        req.method == SIP_METH_BYE) {
       // seems that we have already sent/received a BYE
       // -> we'd better terminate this ASAP
